script.js

```diff
@@ -1,4 +1,704 @@
+    //   { name: "ace", suit: "diamonds", rank: 1 },
+    // ];
+    // dealerHand = [
+    //   { name: "ace", suit: "spades", rank: 1 },
+    //   { name: 10, suit: "spades", rank: 10 },
+    // ];
+    // console.log(currentGameMode);
+    var playerHasBlackJack = checkForBlackJack(playerHand);
+    var dealerHasBlackJack = checkForBlackJack(dealerHand);
+
+    // console.log("player has blackjack? ");
+    // console.log(playerHasBlackJack);
+    // console.log("dealer has blackjack?");
+    // console.log(dealerHasBlackJack);
+
+    // check whether blackjack situation or not
+    if (playerHasBlackJack == true || dealerHasBlackJack == true) {
+      var playerHandTotalValue = calculateTotalHandValue(playerHand);
+      var dealerHandTotalValue = calculateTotalHandValue(dealerHand);
+      // both player have blackjacks -> tie
+      if (playerHasBlackJack == true && dealerHasBlackJack == true) {
+        myOutputValue =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          "You both has blackjack! It is a tie" +
+          displayTotalValue(playerHandTotalValue, dealerHandTotalValue);
+      }
+      // player has blackjack -> player wins
+      else if (playerHasBlackJack == true && dealerHasBlackJack == false) {
+        myOutputValue =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          "Player, You win! You have a blackjack" +
+          displayTotalValue(playerHandTotalValue, dealerHandTotalValue) +
+          "<br>" +
+          imageThumbsUp;
+      }
+      // dealer has blackjack -> dealer wins
+      else if (playerHasBlackJack == false && dealerHasBlackJack == true) {
+        myOutputValue =
+          displayPlayerAndDealerHands(playerHand, dealerHand) +
+          "Opppsy, you lose! Dealer got a black jack!" +
+          displayTotalValue(playerHandTotalValue, dealerHandTotalValue) +
+          "<br>" +
+          imageLost;
+      }
+      console.log(myOutputValue);
+    } else {
+      // playerHand = [
+      //   { name: "queen", suit: "clubs", rank: 12 },
+      //   { name: 5, suit: "diamonds", rank: 5 },
+      // ];
+      // dealerHand = [
+      //   { name: "queen", suit: "spades", rank: 12 },
+      //   { name: 10, suit: "spades", rank: 10 },
+      // ];
+      myOutputValue = "There is no blackjack! Let's calculate";
+      // no blackjack -> check the total value
+      // calculate the total hand value for both dealer and player
+      // console.log("pre function");
+      var playerHandTotalValue = calculateTotalHandValue(playerHand);
+      var dealerHandTotalValue = calculateTotalHandValue(dealerHand);
+      // console.log("post function");
+
+      // console.log("player total hand value");
+      // console.log(playerHandTotalValue);
+      // console.log("dealer total hand value");
+      // console.log(dealerHandTotalValue);
+
+      // compare total hand value
+      // BUST 1 : player total value goes above 21
+      if (playerHandTotalValue > 21) {
```

Comment:
checking if a hand is above 21 is a common occurrence in Blackjack. Hence, creating a function for that will be useful